content/en/blogs/devops-automatic-testing.md

```diff
@@ -0,0 +1,247 @@
+
+The advantage of API testing under the architecture of front-end and back-end seperation lies in the relatively complete API-related documents and materials and the team members' relative familiarity with the API, which will be helpful for the testing. 
+
+The following is an example of running an automated API testing in a pipeline using Postman.
+
+```
+pipeline {
+  agent {
+    kubernetes {
+      label 'apitest'
+      yaml '''apiVersion: v1
+kind: Pod
+spec:
+  containers:
+  - name: newman
+    image: postman/newman_alpine33
+    command: [\'cat\']
+    tty: true
+    volumeMounts:
+    - name: dockersock
+      mountPath: /var/run/docker.sock
+    - name: dockerbin
+      mountPath: /usr/bin/docker
+  volumes:
+  - name: dockersock
+    hostPath:
+      path: /var/run/docker.sock
+  - name: dockerbin
+    hostPath:
+      path: /usr/bin/docker
+      '''
+      defaultContainer 'newman'
+    }
+  }
+
+  parameters {
+    string(name: 'HOST', defaultValue: '10.10.10.10', description: '')
+    string(name: 'PORT', defaultValue: '8000', description: '')
+    string(name: 'USERNAME', defaultValue: 'admin', description: '')
+    string(name: 'PASSWORD', defaultValue: 'password', description: '')
+
+  }
+
+  stages {
+    stage('testing') {
+      steps {
+          sh '''
+          apk add --no-cache bash git openssh
+          git clone https://yourdomain.com/ns/ks-api-test.git
+
+          cd ks-api-test
+
+          sed -i "s/__HOST__/$HOST/g" postman_environment.json
+          sed -i "s/__PORT__/$PORT/g" postman_environment.json
+          sed -i "s/__USERNAME__/$USERNAME/g" postman_environment.json
+          sed -i "s/__PASSWORD__/$PASSWORD/g" postman_environment.json
+
+          npm install -g newman-reporter-htmlextra
+          newman run iam/postman_collection.json -e postman_environment.json -r htmlextra
+          '''
+      }
+    }
+  }
+  post {
+    always {
+        archiveArtifacts 'ks-api-test/newman/*'
+    }
+  }
+}
+```
+
+The artifact will be archived after the running process:
+
+![api-test](/images/blogs/en/devops-automatic-testing/api-test.png)
+
+Download the archived artifact and decompress it to view the testing report:
+
+![](https://pek3b.qingstor.com/kubesphere-docs/png/20200428175325.png)
+
+The framework of automated API testing is easy to establish with the implementation of the following functions:
+
+- Interface Request
+- Response Assertion
+- Request Orchestration
+- Report Generation
+
+It should be ensured that the solution is suitable for the team's API testing and delivery habits. You can develop the solution on your own, or just use the existing tools. The above solution chooses Postman and Newman for the reason that the team generally uses Postman for API testing.
```

Comment:
It should be ensured that the solution is suitable for the team's API testing and delivery habits. You can develop the solution on your own, or just use the existing tools.
->
Keep in mind that you need to select the solution that best suits your team based on your API tests and delivery habits. For example, you can develop your own solution or just use the existing tools.